polygeist/test/polygeist-opt/polygeist-to-llvm.mlir

```diff
@@ -573,3 +541,56 @@ func.func private @view(%arg0: memref<8xi8, 3>, %arg1: index) -> memref<2xf32, 3
   %res = memref.view %arg0[%arg1][] : memref<8xi8, 3> to memref<2xf32, 3>
   return %res : memref<2xf32, 3>
 }
+
+// -----
+
+// CHECK:         llvm.func @malloc(i64) -> !llvm.ptr
+func.func private @malloc(i64) -> !llvm.ptr
+
+// CHECK-LABEL:   llvm.func @f0(
+// CHECK-SAME:                  %[[VAL_0:.*]]: i64) -> !llvm.ptr
+// CHECK:           %[[VAL_2:.*]] = llvm.mlir.null : !llvm.ptr
+// CHECK:           %[[VAL_3:.*]] = llvm.getelementptr %[[VAL_2]]{{\[}}%[[VAL_0]]] : (!llvm.ptr, i64) -> !llvm.ptr, i8
+// CHECK:           %[[VAL_4:.*]] = llvm.ptrtoint %[[VAL_3]] : !llvm.ptr to i64
+// CHECK:           %[[VAL_5:.*]] = llvm.call @malloc(%[[VAL_4]]) : (i64) -> !llvm.ptr
+// CHECK:           llvm.return %[[VAL_5]] : !llvm.ptr
+// CHECK:         }
+func.func private @f0(%size: index) -> memref<?xi8> {
+  %res = memref.alloc(%size) : memref<?xi8>
+  return %res : memref<?xi8>
+}
+
+// CHECK-LABEL:   llvm.func @f1(
+// CHECK-SAME:                  %[[VAL_0:.*]]: i64) -> !llvm.ptr
+// CHECK:           %[[VAL_1:.*]] = llvm.call @malloc(%[[VAL_0]]) : (i64) -> !llvm.ptr
+// CHECK:           llvm.return %[[VAL_1]] : !llvm.ptr
+// CHECK:         }
+func.func private @f1(%size: i64) -> !llvm.ptr {
+  %res = func.call @malloc(%size) : (i64) -> !llvm.ptr
```

Comment:
Do we also test the error case somewhere?

Reply:
Conversion patterns are not tested at all in MLIR. Without #11248, this cannot be tested properly in `cgeist` frontend... OK if I add the test as part of #11248 ?

Reply:
Yes, sure.